python/tank/authentication/user_impl.py

```diff
@@ -206,7 +206,19 @@ def __init__(self, host, login, session_token, http_proxy, password=None, cookie
         self._login = login
         self._session_token = session_token
         self._cookies = cookies
-        self._saml_expiration = saml_expiration
+
+        # saml expiration is meaningful only with SSO enabled.
+        # The presence of cookies implies potential SSO usage.
+        if self._cookies is None:
+            self._saml_expiration = None
+        elif saml_expiration is None:
+            # For lack of accurate information, we will indicates that the
```

Comment:
Why do you lack accurate information? When does that happen. Some notes on why would be appreciated.